MetaMask/metamask-extension
# Fix gas calculation checking wrong account balance

## **Description**

Dapps can connect to / prompt transactions from a different account than the one currently selected within the wallet itself.

When this occurs, `useGasFeeErrors` was doing a balance check against the wrong account.  It was using the current account within the wallet, instead of the account issuing the transaction.  This can cause balance errors even when the account has sufficient funds.

This shows in 2 places in the UI, which are the only places checking `balanceError`:
- Token approvals
- Customizing gas popup

Solution: Check balance of the account issuing the transaction.

## **Manual testing steps**

1. Within the wallet, select an empty account.
2. On the E2E Test Dapp, connect to a funded account.
3. Send ETH
4. Click "🌐 Site suggested" to customize gas
5. A balance error used to appear, but no longer should.

## **Screenshots/Recordings**

### **Before**

https://github.com/MetaMask/metamask-extension/assets/3500406/f48eff21-f2ff-4d96-ba4a-ab15427092ec

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/20770

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
